main-application/image-processing_tasks.c

```diff
@@ -0,0 +1,105 @@
+
+/**
+* @brief prepare the image for processing
+*
+* This task will grab a full image from the cubesense memory and initalize
+* image download and check the results. If we want the image to be a thumbnail
+* it will go through a quality check. The end result will give a compressed file
+* that will be prepared and sent to the downlink manager.
+* @header software-and-command/main-application
+* @param parameters unused
+* @pre None
+* @post Stores the image into the memory storage
+* @return None
+*/
+
+void imagePreperationTask(void* pvParameters){
+    uint_8 imageDownload;
+    uint_8 version
+    unit_8 img = initalizeImageDownload()
+    uint_8 compressed
+    uint_8 check = qualityCheck(img)
+
+    QueueHandle_t xQueueImageVersion;
+    xQueueImageVersion = xQueueCreate(MAX_LENGTH, 1)
+
+    while(1){
+
```

Comment:
- What does this task block/delay on? Right now it will always run and never yield or block
- Does `qualityCheck()` never get called again after this task starts? (which only happens once, remember)